cpp/src/rolling/grouped_rolling.cu

```diff
@@ -371,26 +373,27 @@ std::unique_ptr<column> time_range_window_ASC(column_view const& input,
 /// Each group in the input timestamp column must be sorted,
 /// with null values clustered at either the start or the end of each group.
 /// If there are no nulls for any given group, (nulls_begin, nulls_end) == (0,0).
-std::tuple<rmm::device_vector<size_type>, rmm::device_vector<size_type>>
+std::tuple<rmm::device_uvector<size_type>, rmm::device_uvector<size_type>>
 get_null_bounds_for_timestamp_column(column_view const& timestamp_column,
-                                     rmm::device_uvector<size_type> const& group_offsets)
+                                     cudf::device_span<size_type const> group_offsets,
+                                     rmm::cuda_stream_view stream)
 {
   // For each group, the null values are themselves clustered
   // at the beginning or the end of the group.
   // These nulls cannot participate, except in their own window.
 
   // If the input has n groups, group_offsets will have n+1 values.
   // null_start and null_end should eventually have 1 entry per group.
-  auto null_start = rmm::device_vector<size_type>(group_offsets.begin(), group_offsets.end() - 1);
-  auto null_end   = rmm::device_vector<size_type>(group_offsets.begin(), group_offsets.end() - 1);
+  auto null_start = cudf::detail::make_device_uvector_async(group_offsets, stream);
+  auto null_end   = cudf::detail::make_device_uvector_async(group_offsets, stream);
```

Comment:
I see the -1 removal here is compensated by the device_span group_offsets passing 1 less value.
Should the comment here be updated?

Reply:
Also, the lambda below technically accesses a value outside of the `device_span` by using a data pointer directly, right?
`auto group_end             = d_group_offsets[group_label + 1];`

Perhaps this could use some comments as well.

Reply:
These were good points. I decided that it was too potentially confusing to slice off the last element outside this function, so I reverted that, and tried to clarify both the documentation and the code, along with the optimization suggested by @robertmaynard .